### PR TITLE
add a description of how to compile and run global_oce_cs32

### LINKED ISSUE #23: Update the documentation for global_oce_cs32 adjoint sensitivity run
At present, the README.md file does not contain a description of how to set up and execute the adjoint sensitivity run. Add a set of commands to the README.md file to make this process easier.

### PR DESCRIPTION
**Pull request by mjlosch**
Hi all,

This pull request is an attempt to resolve #23. It is based on previous pull requests by @DanJonesOcean

I am re-doing PR #34 because #34 still cannot be edited.

### CODE REVIEW

**Comment by jm-c** [comment]
@mjlosch I think there are confusing instructions there (regarding forward vs adjoint build and run) that could be improved.
And I would recommend to mention "prepare_run" instead of this long set of symbolic links. "prepare_run" is not only relatively standard accross experiments, it's also documented in the manual and it helps keeping the instructions up-to-date in case some files are moved.

**Comment by mjlosch** [comment]
@jm-c 
could you be a bit more specific about what the confusing instructions, maybe suggest a structure or so? Otherwise this could end up in an endless loop of me trying something and not addressing the point you want to make. 
Here are my suggestion: 
1. we could split "compile the code" into to parts: (1) forward (2) AD-compilation (including the TAF information).
2. prepare_run is definitely a useful script but testreport does more than this (at least judging from the results): I think it does this: ```ln -s ../input_ad.sens/* . && ./prepare_run```; I can add a separate set of commands like this and refer to the manual. I guess, @DanJonesOcean meant this to be a "cut and paste" instruction and we should maybe not make it too complicated?

File: global_oce_cs32/README.md [hunk 1]
```diff
@@ -8,17 +7,55 @@ git clone https://github.com/MITgcm/verification_other
...
+# compile the code
+
+[Note that building the AD-executable requires access to [TAF](http://www.fastopt.de).]
+
+The following steps are basically what `../verification/testreport -t global_oce_cs32 -adm` does if executed in directory `verification_other`.
 
-ln -s ../verification_other/global_oce_cs32 .
 ```
+cd build
+ ../../../tools/genmake2 -mods ../code_ad ${-of your_build_options_file} ${other options, e.g. -nocat4ad}
```
Comment (jm-c) [inline global_oce_cs32/README.md:22]: code_ad is just a link (to get "testreport -adm" to work); sugest to replace it with real dir "code".

File: global_oce_cs32/README.md [hunk 2]
```diff
@@ -8,17 +7,55 @@ git clone https://github.com/MITgcm/verification_other
...
+make CLEAN
+make depend
+make
```
Comment (jm-c) [inline global_oce_cs32/README.md:25]: testreport command above has option "-adm" which will not trigger a simple "make" but instead a "make adall"

File: global_oce_cs32/README.md [hunk 3]
```diff
@@ -8,17 +7,55 @@ git clone https://github.com/MITgcm/verification_other
...
+
+```
+
+The final `make` command will build `mitgcmuv` for forward integrations. For the AD-executable `mitgcmuv_ad`, you have to do
+
+```
+make adall
+```
+
+which will call `taf` (after you have paid for it) to generate the AD-code and then compile.
```
Comment (jm-c) [inline global_oce_cs32/README.md:35]: suggest replacing "(after you have paid for it)" with the need to get license for TAF with a link to FastOpt

**Review by jm-c — COMMENTED** [review]
I think it would be useful to explain how to compile and run the forward set-up first (does not require TAF license); and in second part how to compile mitgcmuv_ad (can re-use same Makefile and build dir) and run the 2 Ad set-up. The instructions how to run each of them is different (since the first input* dir to link files from is different).

**Comment by jrscott** [comment]
I think this is getting there... but in "compile the code" I would just include a more informal syntax:
 ../../../tools/genmake2 -mods ../code -of your_build_options_file [-other options, e.g. -nocat4ad]
 (the $ is confusing)

And I would not mention testreport until after the command. I could not follow what you were suggesting about the testreport directory.

when you think this is good to go let me know and I'll test it

**Comment by jm-c** [comment]
@mjlosch I am going to push some edits (some will need some tweaking).

**Comment by jm-c** [comment]
@mjlosch I made few edits. If you feel some are not improving the clarity you can revert them. And I did not try to add @jrscott suggestions.

**Comment by mjlosch** [comment]
I change the informal syntax, but kept the sentences with testreport. I am fine with it as it is.

**Comment by jrscott** [comment]
why is testreport mentioned at all, and just for the compilation? (testreport normally runs the code as well).
What point are we trying to make with the testreport mention? Is there something special here about what testreport is doing? Otherwise this seems an aside comment, that as located, is merely potentially confusing (e.g. see the support thread this week about big endian file spec; extraneous info just confuses people). Maybe I'm missing something fundamental here, but the purpose of this mention is not apparent to me. If there is a good reason, please explain and I'll help make this more clear. Also "-t" option is not listed in the manual (just -tdir); maybe that should be fixed up in the manual.

I think more useful "extra explanation", for example, might be to mention that prepare_script is a oft-present script in input (or input_ad)  directory that prepares/copies the necessary support files into place. If it is doing something unique here, describe.

**Comment by jm-c** [comment]
I think mentioning testreport is useful for (at least) 2 reasons:
1) it's what Gael was recommending, and a) he is the original author of this set-up and b) He might have some other pieces of doc
   that mentioned testreport so would be useful not to add confusion there.
2) not all set-up in verification_other are tested every night but this one is (both Fwd & Adm).
Will add (back) something later about testreport there.

**Comment by jm-c** [comment]
@jrscott regarding your comment about "prepare_script" (I think you meant "prepare_run" ?):
From my point of view, these "prepare_run" scripts are simple enough that just reading it provides a good idea of what it's doing, and I don't see much of the need to add much more explanations about "prepare_run" in this README.

**Comment by jrscott** [comment]
> @jrscott regarding your comment about "prepare_script" (I think you meant "prepare_run" ?):
> From my point of view, these "prepare_run" scripts are simple enough that just reading it provides a good idea of what it's doing, and I don't see much of the need to add much more explanations about "prepare_run" in this README.

 I meant this more as a general  example as opposed to specific suggestion here, although I would argue that a new user might not immediately realize what "magic" prepare_run might be doing. But really I was arguing more that if the setup required non-standard steps that prepare_run was doing, it might be worth mentioning. Along the lines of including other info in the readme that might be useful to convey, along the lines of what you are suggesting in your point #2.

**Comment by jrscott** [comment]
To clarify what I'm suggesting: right now there are three main sections "basic instructions to install", "basic instructions to compile", and "basic instructions to run"; info about whether/how the setup is tested in testreport falls outside these three, perhaps a new section about testing procedure or other relevant info etc. Or let me know if I'm still not understanding your objective.

**Comment by jm-c** [comment]
I've just added few lines ; I think we could add more information related to the use of testreport (also could mention where ans which
reference output to compare with) but it does not seem strictly necessary to me.

@jrscott regarding the general purpose of this README, we could limit the content of this file to just instructions that are specific to this set-up, and in this case this PR is of no use (the original README is good enough); or we could add more instructions and details about this set-up. This is more in line with many README that are found in many verification experiments (although not up-to-date), for instance in global_ocean.90x40x15, global_ocean.cs32x15 or so_box_biogeo.

**Comment by jrscott** [comment]
LGTM

**Pull request merged**